hpy/devel/include/hpy/cpython/misc.h

```diff
@@ -276,6 +276,32 @@ HPyAPI_FUNC HPy HPy_CallTupleDict(HPyContext *ctx, HPy callable, HPy args, HPy k
     return ctx_CallTupleDict(ctx, callable, args, kw);
 }
 
+#if PY_VERSION_HEX < 0x03090000
+#define PyObject_Vectorcall _PyObject_Vectorcall
+#endif
+
+HPyAPI_FUNC HPy HPy_Call(HPyContext *ctx, HPy callable, const HPy *args, size_t nargs, HPy kwnames)
+{
+    if (sizeof(HPy) == sizeof(PyObject *)) {
+        return _py2h(PyObject_Vectorcall(_h2py(callable), (PyObject *const *)args, nargs, _h2py(kwnames)));
+    }
+    return ctx_Call(ctx, callable, args, nargs, kwnames);
+}
```

Comment:
I'm confused by this check. Can it ever happen that `sizeof(HPy)` != `sizeof(PyObject *)`?
There is already code around which relies on the fact that in CPython mode `HPy` is basically a `PyObject *` in disguise: so if this check is needed, then we have lots of problems everywhere else.
If it's not needed, then we should kill it.